### PR TITLE
Filters - num of availble addresses in CIDR IPv6 / IPv4

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22805
fixes: https://github.com/demisto/etc/issues/23309

### Filters for CIDR IPv6 or IPv4:
1. **Greater** - Check whether CIDR has more available addresses than a given number.
2. **Lower** - Check whether CIDR has less available addresses than a given number.

### General command for networks
3. **IPNetwork command** - Return overall details about Network - IPv4 /IPv6, So the sec team will not need to wait for our development, I exposed global details regarding the network, which can be manipulated and used - examples for Entry context:
```json
'Network : [{
  "Broadcast_address": "192.168.0.255",
  "CIDR": "192.168.0.4/24",
  "First_address": "192.168.0.0",
  "Last_address": "192.168.0.255",
  "Max_prefix_len": 32,
  "Num_addresses": 256,
  "Private": true,
  "Version": 4
},
{
  "Broadcast_address": "2002::1234:abcd:ffff:c0a8:101",
  "CIDR": "2002::1234:abcd:ffff:c0a8:101/127",
  "First_address": "2002::1234:abcd:ffff:c0a8:100",
  "Last_address": "2002::1234:abcd:ffff:c0a8:101",
  "Max_prefix_len": 128,
  "Num_addresses": 2,
  "Private": false,
  "Version": 6
}
]
```

## Screenshots
image

## Minimum version of Demisto
- [ ] 4.5.0
- [x] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

**Comment by GalRabin** [comment]
@David-BMS approved the functionality already.

**Comment by GalRabin** [comment]
@kirbles19  I will appreciate your review:
1. Lower:
YML
Description
2. Greater:
YML
Description
3.. General command:
YML
Description

File: Packs/Legacy/Scripts/GreaterCidrNumAddresses/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+- Check if number of availbe addresses is lower than given numebr.
+
+## [19.11.1] - 2019-11-26
+These notes will be published in the next release notes
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/GreaterCidrNumAddresses/CHANGELOG.md:5]: remove

File: Packs/Legacy/Scripts/IPNetwork/IPNetwork_description.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## CIDR info - IPv6/IPv4
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/IPNetwork/IPNetwork_description.md:1]: only integrations have a description file, move to the script comment

File: Packs/Legacy/Scripts/LowerCidrNumAddresses/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+- Check if number of availbe addresses is lower than given numebr.
+
+## [19.11.1] - 2019-11-26
+These notes will be published in the next release notes
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/LowerCidrNumAddresses/CHANGELOG.md:5]: remove

File: Packs/Legacy/Scripts/IPNetwork/IPNetwork.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        ec,
+        ""
+    )
+
+
+def main():
+    try:
+        return_outputs(*cidr_command(demisto.args()))
+    except Exception as ex:
+        return_error(f'Failed to execute HelloWorldScript. Error: {str(ex)}')
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/IPNetwork/IPNetwork.py:66]: fix error

File: Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    ip_cidr_obj = ipaddress.ip_network(address=ip_cidr, strict=False)
+    if ip_cidr_obj.num_addresses < int(max_num_addresses):
+        return True
+    return False
+
+
+def main():
+    ip_cidrs = argToList(demisto.args()['left'])
+    max_num_addresses = argToList(demisto.args()['right'])[0]
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses.py:25]: why are you ignoring all but the first?

File: Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses_description.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+## Num of addresses is lower than availble in IPv4 CIDR
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses_description.md:1]: same

File: Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses_description.md [hunk 7]
```diff
@@ -0,0 +1,2 @@
+## Num of addresses is greater than availble in IPv4 CIDR
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses_description.md:1]: same

File: Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    ip_cidr_obj = ipaddress.ip_network(address=ip_cidr, strict=False)
+    if ip_cidr_obj.num_addresses > int(min_num_addresses):
+        return True
+    return False
+
+
+def main():
+    ip_cidrs = argToList(demisto.args()['left'])
+    min_num_addresses = argToList(demisto.args()['right'])[0]
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses.py:25]: why are you ignoring all but the first?

File: Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses.py (hunk 5 above)
Comment (GalRabin) [reply Packs/Legacy/Scripts/LowerCidrNumAddresses/LowerCidrNumAddresses.py:25]: My mistake

File: Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses.py (hunk 8 above)
Comment (GalRabin) [reply Packs/Legacy/Scripts/GreaterCidrNumAddresses/GreaterCidrNumAddresses.py:25]: My mistake

File: Packs/Legacy/Scripts/IPNetwork/IPNetwork.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+# Gather information regarding CIDR -
+#     1. Broadcast_address
+#     2. CIDR
+#     3. First_address
+#     4. Last address
+#     5. Max prefix len
+#     6. Num addresses
+#     7. Private
+#     8. Version
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/IPNetwork/IPNetwork.py:18]: I meant the `comment` key in the yml which acts as the script descriptions in Demisto.

File: Tests/secrets_white_list.json [hunk 10]
```diff
@@ -21,6 +21,10 @@
   ],
   "iocs": {
     "ips": [
+      "192.168.0.4",
```
Comment (yaakovi) [inline Tests/secrets_white_list.json:24]: not relevant for packs.
should be in `.secret_ignore` of the pack

**Pull request merged**